csi/src/util.rs

```diff
@@ -215,6 +216,37 @@ pub fn spawn_grpc_task<R: Send + 'static>(
     .detach();
 }
 
+/// Send async failure `gRPC` response
+pub async fn async_fail_test<R>(sink: UnarySink<R>, rsc: RpcStatusCode, err: &DatenlordError) {
+    debug_assert_ne!(
+        rsc,
+        RpcStatusCode::OK,
+        "the input RpcStatusCode should not be OK"
+    );
+    let rs = RpcStatus::new(rsc, Some(format!("{}", err)));
```

Comment:
Original err string will contain the context message of anyhow. If we use our error type, the error message will lose context information. Is this OK?

Reply:
Or we need add error message to our error code?